### PR TITLE
Test conditionally forward filling `default.vw_pin_universe`

### PR DESCRIPTION
**Pull request by wrridgeway**
Reporting on 2024 is pretty difficult with no 2024 spatial data. Let's try forward filling it in `vw_pin_universe` and see what that gets us.

### CODE REVIEW

**Comment by wrridgeway** [comment]
## Testing

This doesn't seem to break anything. Here's a row count for old vs new by year:

```
WITH new AS (
select year, count(*) as new_count
from "z_ci_415-test-conditionally-forward-filling-defaultvw-pin-universe_default".vw_pin_universe
group by year
),
old AS (
select year, count(*) as old_count
from "default".vw_pin_universe
group by year
)

select
    old.year,
    old.old_count,
    new.new_count
from old
left join new
    on old.year = new.year
```

year | old_count | new_count
-- | -- | --
2024 | 1863492 | 1863492
2023 | 1864161 | 1864161
2022 | 1865018 | 1865018
2021 | 1864589 | 1864589
2020 | 1864035 | 1864035
2019 | 1864102 | 1864102
2018 | 1864590 | 1864590
2017 | 1864621 | 1864621
2016 | 1863996 | 1863996
2015 | 1862756 | 1862756
2014 | 1862170 | 1862170
2013 | 1861935 | 1861935
2012 | 1862181 | 1862181
2011 | 1861008 | 1861008
2010 | 1859244 | 1859244
2009 | 1849650 | 1849650
2008 | 1832618 | 1832618
2007 | 1804219 | 1804219
2006 | 1771314 | 1771314
2005 | 1740965 | 1740965
2004 | 1716218 | 1716218
2003 | 1693153 | 1693153
2002 | 1673735 | 1673735
2001 | 1655398 | 1655398
2000 | 1637269 | 1637269
1999 | 1620990 | 1620990

File: aws-athena/views/default-vw_pin_universe.sql [hunk 1]
```diff
@@ -1,4 +1,29 @@
 -- Source of truth view for PIN location
+
+-- This CTE ensures the most recent year of pardat data is joined to spatial
+-- data regardless of timing. This is necessary since spatial data releases late
+-- in a calendar year while the CCAO's universe of PINs is rolled-over much
```
Comment (dfsnow) [inline aws-athena/views/default-vw_pin_universe.sql:5]:
```suggestion
-- in a calendar year while the CCAO's universe of PINs is rolled over much
```

File: aws-athena/views/default-vw_pin_universe.sql [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+-- sooner.
```
Comment (dfsnow) [inline aws-athena/views/default-vw_pin_universe.sql:6]: **suggestion:** Can you add a small comment diagram that shows the missingness problem? I feel like the timing/missingness here is really confusing if you're not intimately familiar with the data release schedule.

File: aws-athena/views/default-vw_pin_universe.sql [hunk 3]
```diff
@@ -1,4 +1,29 @@
...
+            ELSE taxyr
+        END AS join_year,
+        class,
+        nbhd,
+        cur,
+        deactivat
+    FROM {{ source('iasworld', 'pardat') }}
+
+)
+
```
Comment (dfsnow) [inline aws-athena/views/default-vw_pin_universe.sql:26]: **question:** Worth making into its own ephemeral model or view?

**Review by dfsnow — COMMENTED** [review]
@wrridgeway This looks pretty good, but how does it affect views that are downstream of universe? Can you use the dbt lineage graph to grab and diff the downstream views? Would be useful to see an `EXCEPT` test on all of them.

File: aws-athena/views/default-vw_pin_universe.sql [hunk 4]
```diff
@@ -123,18 +149,18 @@ SELECT
     vwl.access_cmap_walk_data_year,
     vwl.misc_subdivision_id,
     vwl.misc_subdivision_data_year
-FROM {{ source('iasworld', 'pardat') }} AS par
+FROM pardat_adjusted_years AS par
 LEFT JOIN {{ source('iasworld', 'legdat') }} AS leg
     ON par.parid = leg.parid
     AND par.taxyr = leg.taxyr
```
Comment (wrridgeway) [inline aws-athena/views/default-vw_pin_universe.sql:155]: This should still use `taxyr` for joining.

**Comment by wrridgeway** [comment]
> @wrridgeway This looks pretty good, but how does it affect views that are downstream of universe? Can you use the dbt lineage graph to grab and diff the downstream views? Would be useful to see an `EXCEPT` test on all of them.

Nothing is downstream of vw_pin_universe.

File: aws-athena/views/default-vw_pin_universe.sql (hunk 3 above)
Comment (wrridgeway) [reply aws-athena/views/default-vw_pin_universe.sql:31]: Our naming syntax for symlinks in dbt/models/ makes ethereal tables error out. Probably not worth working around ATM.

**Review by dfsnow — APPROVED** [review]
Alright then @wrridgeway, this looks ready to go.

**Pull request merged**